File: app/src/main/java/com/wkz/viewpager/transformer/BackgroundToForegroundTransformer.java
package com.wkz.viewpager.transformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * @desc 后景到前景的转换效果
 * @date 2017-06-30
 */
public class BackgroundToForegroundTransformer extends AbstractBaseTransformer {
    @Override
    protected void onTransform(View page, float position) {
        float pageWidth = page.getWidth();
        float pageHeight = page.getHeight();
        float scale = max(position < 0 ? 1F : Math.abs(1F - position), 0.25F);

        ViewHelper.setScaleX(page, scale);
        ViewHelper.setScaleY(page, scale);
        ViewHelper.setPivotX(page, pageWidth * 0.5F);
        ViewHelper.setPivotY(page, pageHeight * 0.5F);
        ViewHelper.setTranslationX(page, position < 0 ? pageWidth * position : -pageWidth * position * 0.25F);
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/transformer/BackgroundToForegroundTransformer.kt
package com.fphoenixcorneae.transformer

import android.view.View
import kotlin.math.abs

/**
 * @desc 后景到前景的转换效果
 * @date 2017-06-30
 */
class BackgroundToForegroundTransformer : AbstractBaseTransformer() {

    override fun onTransform(page: View, position: Float) {
        val pageWidth = page.width.toFloat()
        val pageHeight = page.height.toFloat()
        val scale = max(
            if (position < 0) {
                1f
            } else {
                abs(1f - position)
            },
            0.25f
        )
        page.apply {
            scaleX = scale
            scaleY = scale
            pivotX = pageWidth * 0.5f
            pivotY = pageHeight * 0.5f
            translationX = if (position < 0) pageWidth * position else -pageWidth * position * 0.25f
        }
    }
}